opentelemetry-otlp/src/exporter/http/mod.rs

```diff
@@ -410,4 +451,21 @@ mod tests {
             // You may also want to assert on the specific error type if applicable
         });
     }
+
+    #[test]
+    fn test_add_header_from_string() {
```

Comment:
as a future improvement, we need to vastly expand the test coverage and make sure all priority orders are tested, and the bad input behaviors are covered as well..

Reply:
Good point. Improved test coverage, with priority order and bad input.